src/evm.rs

```diff
@@ -109,11 +128,90 @@ impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> EvmNamespaceT
+        let inner = Arc::clone(&self.node);
+        Box::pin(async move {
+            match inner.write() {
+                Ok(mut inner) => {
+                    let (keys, block, bytecodes) = {
+                        let mut storage_view = StorageView::new(&inner.fork_storage);
+                        let mut oracle_tools = OracleTools::new(&mut storage_view, HistoryEnabled);
+
+                        let bootloader_code = &inner.system_contracts.baseline_contracts;
+                        let block_context = inner.create_block_context();
+                        let block_properties =
+                            InMemoryNodeInner::<S>::create_block_properties(bootloader_code);
+
+                        let block = Block {
+                            hash: H256::random(),
+                            number: inner.current_miniblock.saturating_add(1).into(),
+                            timestamp: block_context.block_timestamp.into(),
+                            l1_batch_number: Some(block_context.block_number.into()),
+                            ..Default::default()
+                        };
+
+                        // init vm
+                        let mut vm = init_vm_inner(
+                            &mut oracle_tools,
+                            BlockContextMode::NewBlock(block_context.into(), Default::default()),
+                            &block_properties,
+                            BLOCK_GAS_LIMIT,
+                            bootloader_code,
+                            TxExecutionMode::VerifyExecute,
+                        );
+
+                        vm.execute_till_block_end(BootloaderJobType::BlockPostprocessing);
+
+                        let bytecodes = vm
+                            .state
+                            .decommittment_processor
+                            .known_bytecodes
+                            .inner()
+                            .clone();
+
+                        let modified_keys = storage_view.modified_storage_keys().clone();
+                        (modified_keys, block, bytecodes)
+                    };
+
+                    for (key, value) in keys.iter() {
+                        inner.fork_storage.set_value(*key, *value);
+                    }
+
+                    // Write all the factory deps.
+                    for (hash, code) in bytecodes.iter() {
+                        inner.fork_storage.store_factory_dep(
+                            u256_to_h256(*hash),
+                            code.iter()
+                                .flat_map(|entry| {
+                                    let mut bytes = vec![0u8; 32];
+                                    entry.to_big_endian(&mut bytes);
+                                    bytes.to_vec()
+                                })
+                                .collect(),
+                        )
+                    }
+                    println!("👷 Mined block #{}", block.number.as_u64());
```

Comment:
Please update this to use the new logger
```suggestion
                    log::info!("👷 Mined block #{}", block.number.as_u64());
```

P.S. Love the little construction worker emoji 👍